lib/facility.rb

```diff
@@ -1,14 +1,50 @@
+
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles, 
+              :collected_fees,
+              :plate_type
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+    @plate_type = :regular
+  end
+
+  def add_service(service) # adds a service (passed as an arguement into the service parameter) 
+    @services << service # into the services instance variable, which is an array
   end
 
-  def add_services(service)
-    @services << service
+  def register_vehicle(vehicle) # add the registration date 6:46 pm
+    if services.include?("Vehicle Registration")
+      if vehicle.antique?
+        @collected_fees += 25
+        @registered_vehicles << vehicle
+        vehicle.plate_type
+        vehicle.assign_date
+      elsif vehicle.electric_vehicle?    
+        @collected_fees += 200 
+        @registered_vehicles << vehicle
+        vehicle.plate_type
+        vehicle.assign_date
+      else
+        @collected_fees += 100
+        @registered_vehicles << vehicle
+        vehicle.plate_type
+        vehicle.assign_date
+      end
+    end  
```

Comment:
Yes, so all the logic here is working great. We could refactor this be noticing that every branch of logic is doing `vehicle.plate_type` and `vehicle.assign_date` which means those two lines could just be above line 29, before your second if statement. 
Great job making helper methods of `plate_type` and `assign_date`!